src/kafkerl_protocol.erl

```diff
@@ -150,17 +150,10 @@ parse_fetch_response(Bin, {Remainder, CorrelationId, Steps}) ->
 parse_metadata_response(<<CorrelationId:?UINT,
                           BrokerCount:?UINT,
                           BrokersBin/binary>>) ->
-  case parse_brokers(BrokerCount, BrokersBin) of
-    {ok, Brokers, <<TopicCount:?UINT, TopicsBin/binary>>} ->
-      case parse_topic_metadata(TopicCount, TopicsBin) of
-        {ok, Metadata} ->
-          {ok, CorrelationId, {Brokers, Metadata}};
-        {error, _Reason} = Error ->
-          Error
-      end;
-    {error, _Reason} = Error ->
-      Error
-  end;
+  {ok, Brokers, <<TopicCount:?UINT, TopicsBin/binary>>} =
```

Comment:
We can't just let it crash here, because an error is a perfectly valid kafka response. Basically, it's a response that points to some error like the partition not existing and such that should be handled differently.